### PR TITLE
images: Update LLVM to 17.0.6

### PR DESCRIPTION
**Pull request by gentoo-root**
It's finally happening!

Preceding BPF fixes: https://github.com/cilium/cilium/pull/31403, https://github.com/cilium/cilium/pull/31459, https://github.com/cilium/cilium/pull/31849

```release-note
images: Update LLVM to 17.0.6
```

### CODE REVIEW

**Review by borkmann — APPROVED** [review]

File: .github/actions/unit-tests/prepare.sh
Comment (YutaroHayakawa) [inline .github/actions/unit-tests/prepare.sh:1]: Why did they stop signing? Don't we have any alternatives?

**Review by YutaroHayakawa — APPROVED** [review]
LGTM with a question.

File: .github/actions/unit-tests/prepare.sh
Comment (brlbil) [reply .github/actions/unit-tests/prepare.sh:1]: It seems signatures have been missing since version 14.0.1

**Review by brlbil — APPROVED** [review]

File: .github/actions/unit-tests/prepare.sh
Comment (gentoo-root) [reply .github/actions/unit-tests/prepare.sh:1]: I couldn't find the reason why they stopped signing the binaries. They still sign the sources. I actually wanted to use the cilium-builder image to build everything (https://github.com/cilium/cilium/pull/31754), but some unit tests use the same Clang for both BPF and native build, and cilium-builder only supports BPF (that's intended, to reduce space). I still plan to try to come with some solution as a follow-up.

Another alternative is to build LLVM from signed sources, but it's not feasible in the CI.

One more option is to switch to install-llvm-action that we still use in two other places for the same reason, this way we'll have more consistency, but it doesn't check signatures either.

Comment (gentoo-root) [reply .github/actions/unit-tests/prepare.sh:1]: > switch to install-llvm-action

Implemented this for consistency.

**Review by tklauser — APPROVED** [review]

**Pull request merged**